Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -235,12 +236,44 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
               information += OverCostlyMessage;
             }
           }
+
         }
       }
 
       return information;
     }
 
+    private string GetProofDependencyInformation(IdeState state, TopLevelDeclMemberVerificationTree node, List<AssertionBatchVerificationTree> orderedAssertionBatches) {
+      var allPotentialDependencies = new HashSet<ProofDependency>();
+      var usedDependencies = new HashSet<ProofDependency>();
+      var program = (Program)state.Program;
+      var result = "";
+      if (program is null) {
+        return "null program";
+      }
+
+      var depManager = program.ProofDependencyManager;
+      foreach (var assertionBatch in orderedAssertionBatches) {
+        var batchDependencies = assertionBatch.CoveredIds.Select(depManager.GetFullIdDependency);
+        var batchPotentialDependencies = depManager.GetPotentialDependenciesForDefinition(assertionBatch.VerboseName);
+        allPotentialDependencies.UnionWith(batchPotentialDependencies);
+        usedDependencies.UnionWith(batchDependencies);
+      }
+
+      if (usedDependencies.Any() && options.TrackVerificationCoverage) {
+        result += usedDependencies.Aggregate("\n\nProof dependencies:\n",
+          (acc, dep) => acc + $"* `{dep.RangeString()}: {dep.Description}`\n");
+      }
+
+      var unusedDependencies = allPotentialDependencies.Except(usedDependencies);
+      if (unusedDependencies.Any() && options.TrackVerificationCoverage) {
+        result += unusedDependencies.Aggregate("\n\nUnused program elements:\n",
```

Comment:
It seems strange to report at location X (the method header), about an unused program element that occurs at location Y. Why not only report at location Y ?

Reply:
It depends which question you want to answer:
* Is this element being used? In that case, you want to report that information on hover of location Y. We already do this on assumptions and requires, but not for specific expressions, as they are not explicitly part of the specifications, so I feel it would put too much noise on the body.
* Are there any elements being not used? In that case, you need a place to report that information that does not depend on the elements, so the method header at location X, not depending on any element, is a good choice for expressions without putting too much noise on the body.

Reply:
It might be easier to discuss with screenshots showing this UI